src/service/action-impl.js

```diff
@@ -176,7 +176,15 @@ export class ActionService {
       // fast-click.
       this.root_.addEventListener('click', event => {
         if (!event.defaultPrevented) {
-          this.trigger(dev().assertElement(event.target), 'tap', event);
+          this.trigger(dev().assertElement(event.target), name, event);
+        }
+      });
+      this.root_.addEventListener('keydown', event => {
+        if (event.keyCode == 13 /* enter */ ||
+            event.keyCode == 32 /* space */) {
+          if (!event.defaultPrevented) {
```

Comment:
could we also do `&& this.ampdoc.win.document.activeElement == event.target`? That should prevent `trigger` to run needlessly for every space/enter press.